src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -270,9 +350,45 @@ protected virtual void Dispose(bool disposing)
                     {
                         _channelSource.Dispose();
```

Comment:
This shouldn't be in a catch. Don't we always want to call Dispose on the channelSource?

Reply:
Never mind. It only looked like it was in a catch because github hid some lines from view.